django/template/defaultfilters.py

```diff
@@ -28,10 +29,34 @@
 register = Library()
 
 
+def autoescape_aware(func):
+    """
+    Decorator for filters which should automatically escape text based
+    on whether or not autoescape is True.
+    """
+    def _dec(*args, **kwargs):
+        args = list(args)
+        try:
+            autoescape = kwargs['autoescape']
+        # if value is not defined, get default value
```

Comment:
I'm not following the expected usage here. Surely (?) I'm only applying this to filters `needs_autoescape=True` 

(It seems overly complex…)